wsr-tic-tac-toe/src/redux/reducers/index.ts

```diff
@@ -0,0 +1,137 @@
+                diagonalSet2.add((i + 1) * (boardSize - 1));
+            }
+            let winningScenarios = [];
+            winningScenarios = [
+                ...WSh.map((set) => Array.from(set)),
+                ...WSv.map((set) => Array.from(set)),
+                Array.from(diagonalSet1),
+                Array.from(diagonalSet2),
+            ];
+            return [...winningScenarios];
+        }
+        default:
+            return state;
+    }
+}
+
+const initialBoardState: BoardState = {
+    boardSize: 0,
+    created: false,
+    squares: [],
+    turnOrder: 'player1',
+    gameState: GAME_STATE.ONGOING
+};
+
+export function boardReducer(state = initialBoardState, action: BoardStateActionTypes) {
+    switch (action.type) {
+        case CREATE_BOARD: {
+            let boardSize = action.payload;
+            let initializedSquares: SquareState[] = [];
+            // let squares: SquareState[][] = [];
+            for (let i: number = 0; i < boardSize * boardSize; i++) {
+                initializedSquares = [
+                    ...initializedSquares,
+                    { selected: false, id: i, selectedBy: '' },
+                ]; // creating a 1D array of squares with {selected: false, id: 0,...,boardSize}
+            }
+            return {
+                ...state,
+                boardSize: boardSize,
+                squares: initializedSquares,
+                created: true,
+            };
+        }
+        case REMOVE_BOARD: {
+            return initialBoardState; // return initialBoardState
+        }
+        case CLEAR_BOARD: { // kinda unclear how I should do this with {...state,...} and not Object.assign (though the end result should be the same)
+            return Object.assign(
+                {},
+                state,
+                { turnOrder: 'player1' },
+                { gameState: GAME_STATE.ONGOING},
+                {
+                    squares: state.squares.map((square: SquareState) => {
+                        if (square.selected)
+                            return { ...square, selected: false, selectedBy: false };
+                        else return square;
+                    }),
+                }
+            );
+        }
+        case SELECT_SQUARE: {
+            return Object.assign(
+                {},
+                state,
+                { turnOrder: action.payload.selectedBy === 'player1' ? 'player2' : 'player1' },
+                {
+                    squares: state.squares.map((square: SquareState) => {
+                        if (square.id !== action.payload.id) return square;
+                        else return { ...square, ...action.payload };
+                    }),
+                }
+            );
+        }
+        case SET_GAME_STATE: {
+          return {...state, gameState: action.payload}
+        }
+        default:
+            return state;
+    }
+}
+
+export const rootReducer = combineReducers({
```

Comment:
your reducers themselves are not typed I think, that's why stuff like "false" in a param that has type "string" didn't make TS very angry.
You can either type root reducer by doing 
combinedReducers<AppState>(...)

or type each reducer individually by adding their return type, eg.
const someReducer = (state, action): ReducerType => {...}
or
function someReudcer (state, action): ReducerType {...}